### DISCUSSION TITLE
How do upload 5000+ Images/Assets in Pimcore

### DISCUSSION THREAD

**Discussion by petritz** [pimcore/pimcore#13083] [Q&A / Support]
We are currently trying to get over 5000 images into pimcore without success. Any tried method is always slow and ultimately fails. What worked in the past was uploading the directory and "importing it from server". 

Are there any other and faster possibilities?

Thanks and Regards!

**Comment by BlackbitDevs** [comment]
You can use Data Director bundle for this, please have a look at https://www.youtube.com/watch?v=OJKxWtgwi3c&list=PL4-QRNfdsdKIfzQIP-c9hRruXf0r48fjt&index=3&t=4161s

This way you can either mount your images somewhere on your Pimcore server (or simply upload them somewhere) and then use `File system` as `Source data type`. You can even filter them by certain criteria (file extension, image resulution, image colorspace etc.)

Alternatively you can create a CSV file with all your asset URLs and import the assets from the URLs in the CSV - but of course this will be slower because the files would have to be loaded from the URLs.

The import itself is run in a separate CLI process which will not time out - Pimcore's "import from server" does it in a normal webserver process which may be prone to timing out because of PHP's `max_execution_time` or Apache's `FcgidIOTimeout` or something similar.
